Runtime/Hosting/AssemblyExporter.cs

```diff
@@ -0,0 +1,336 @@
+
+        Trace($"< AssemblyExporter.ExportStaticClass() => [{classProperties.Count}]");
+        return staticClassObject;
+    }
+
+    private JSValue ExportClass(Type classType)
+    {
+        if (_typeObjects.TryGetValue(classType, out JSReference? typeObjectReference))
+        {
+            return typeObjectReference!.GetValue()!.Value;
+        }
+
+        // TODO: Emit static and instance property and method adapter methods,
+        // then define the class.
+
+        return new JSObject();
+    }
+
+    private JSValue ExportStruct(Type structType)
+    {
+        if (_typeObjects.TryGetValue(structType, out JSReference? typeObjectReference))
+        {
+            return typeObjectReference!.GetValue()!.Value;
+        }
+
+        // TODO: Export struct proeprties and methods.
+
+        return new JSObject();
+    }
+
+    private JSValue ExportEnum(Type enumType)
+    {
+        if (_typeObjects.TryGetValue(enumType, out JSReference? typeObjectReference))
+        {
+            return typeObjectReference!.GetValue()!.Value;
+        }
+
+        // TODO: Export enum values as properties on an object.
+
+        return new JSObject();
+    }
+
+    private void EmitStaticMethod(TypeBuilder typeBuilder, MethodInfo method)
+    {
+        MethodBuilder methodBuilder = typeBuilder.DefineMethod(
+            method.Name,
+            MethodAttributes.Public | MethodAttributes.Static | MethodAttributes.Final,
+            returnType: typeof(JSValue),
+            parameterTypes: new[] { typeof(JSCallbackArgs) });
+        ILGenerator il = methodBuilder.GetILGenerator();
+
+        /*
+         * private static JSValue MethodClass_MethodName(JSCallbackArgs __args)
+         * {
+         *     var param0Name = (Param0Type)__args[0];
+         *     ...
+         *     var __result = MethodClass.MethodName(param0, ...);
+         *     return (JSValue)__result;
+         * }
+         */
+
+        var parameters = method.GetParameters();
+        for (int i = 0; i < parameters.Length; i++)
+        {
+            il.DeclareLocal(parameters[i].ParameterType);
+        }
+
+        il.Emit(OpCodes.Nop);
+        for (int i = 0; i < parameters.Length; i++)
+        {
+            EmitLoadAndConvertArg(il, argumentIndex: i, parameters[i].ParameterType, localIndex: i);
+        }
+
+        // Load all the local vars onto the stack.
+        for (int i = 0; i < parameters.Length; i++)
+        {
+            il.Emit(OpCodes.Ldloc, i);
+        }
+        il.Emit(OpCodes.Call, method); // Call the target (static) method!
+
+        EmitConvertAndReturnResult(il, method.ReturnType);
+    }
+
+    private void EmitStaticPropertyGet(TypeBuilder typeBuilder, PropertyInfo property)
+    {
+        // TODO
+    }
+
+    private void EmitStaticPropertySet(TypeBuilder typeBuilder, PropertyInfo property)
+    {
+        // TODO
+    }
+
+    private static void EmitLoadAndConvertArg(
+        ILGenerator il,
```

Comment:
>ILGenerator [](http://example.com/codeflow?start=8&length=11)

I wonder why not use the [https://learn.microsoft.com/en-us/dotnet/api/system.linq.expressions?view=net-7.0](https://learn.microsoft.com/en-us/dotnet/api/system.linq.expressions?view=net-7.0) instead.
I found that generating code this way is much simpler and less error prone. I used to use it in RNW before we switched to compile time code generation (see in the end of file): [https://github.com/microsoft/react-native-windows/blob/main/vnext/Microsoft.ReactNative.Managed/JSValueWriterGenerator.cs](https://github.com/microsoft/react-native-windows/blob/main/vnext/Microsoft.ReactNative.Managed/JSValueWriterGenerator.cs)

Reply:
I'll experiment with that approach.

Reply:
Update: switched to lambda expressions.

It's roughly the same number of lines of code (perhaps more verbose), but I think it is safer compared to using the emit API where it was very easy for a bug to result in invalid IL.